File: 2530-maximal-score-after-applying-k-operations/2530-maximal-score-after-applying-k-operations.cpp
class Solution {
public:
    long long maxKelements(vector<int>& nums, int k) {
        priority_queue<int> pq;
        for(int i=0;i<nums.size();i++){
            pq.push(nums[i]);
        }
        long long ans = 0;
        while(k--){
            ans += pq.top();
            int x = ceil(pq.top()/3.0);
            pq.pop();
            pq.push(x);
        }
        return ans;
    }
};

File: 2530-maximal-score-after-applying-k-operations/2530-maximal-score-after-applying-k-operations.java
class Solution {
    public long maxKelements(int[] nums, int k) {
        PriorityQueue<Integer> pq = new PriorityQueue<>((x,y)->(Integer.compare(y,x)));
        for(int i=0;i<nums.length;i++){
            pq.offer(nums[i]);
        }
        long ans = 0L;
        while(k-- > 0){
            ans += pq.peek();
            int x = (int) Math.ceil(pq.peek() / 3.0);
            pq.poll();
            pq.offer(x);
        }
        return ans;
    }
}